### PR TITLE
front: fix constraint distribution state in simulation settings

### LINKED ISSUE #9811: Regularity margin parameter is set for all trains when you change it
### What happened?

When you change the regularity margin parameter (linear / Mareco), it changes for all trains.

So if we calculate another train, it changes this parameter for the calculus of the train without knowing it.

https://github.com/user-attachments/assets/f1d4baac-1a94-4314-8b9c-9903a6c0c147

### What did you expect to happen?

The parameter should be set individually for each train.

### How can we reproduce it (as minimally and precisely as possible)?

1. Open a scenario with several trains
2. Modify a train and go to Simulation parameters
3. Change the regularity margin (linear / Mareco)
4. Validate the modifications
5. Modify another train
6. Go to simulation parameter and see that the regularity margin is the same

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

6be0604

### PR DESCRIPTION
**Pull request by RomainValls**
close #9811

### CODE REVIEW

**Review by clarani — CHANGES_REQUESTED** [review]
Hum it would be better to fix this directly in the action `selectTrainToEdit` ;) you can get the `constraint_distribution` of the trainSchedule there and directly set the state

File: front/src/applications/operationalStudies/views/ManageTrainSchedule.tsx
Comment (clarani) [inline front/src/applications/operationalStudies/views/ManageTrainSchedule.tsx:1]: You don't need the change of this file anymore ;)

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Review by Math-R — APPROVED** [review]
LGTM !

**Review by Yohh — APPROVED** [review]

**Pull request merged**